vdev/src/testing/runner.rs

```diff
@@ -21,6 +22,35 @@ const TEST_COMMAND: &[&str] = &[
     "--no-fail-fast",
     "--no-default-features",
 ];
+// The upstream container we publish artifacts to on a successful master build.
+const UPSTREAM_IMAGE: &str =
+    "docker.io/timberio/vector-dev:sha-3eadc96742a33754a5859203b58249f6a806972a";
+
+static CONTAINER_TOOL: Lazy<OsString> =
+    Lazy::new(|| env::var_os("CONTAINER_TOOL").unwrap_or_else(detect_container_tool));
+
+fn detect_container_tool() -> OsString {
+    for tool in ["docker", "podman"] {
+        if Command::new(tool)
+            .arg("version")
+            .stdout(Stdio::null())
+            .stderr(Stdio::null())
+            .spawn()
+            .and_then(|mut child| child.wait())
+            .map_or(false, |status| status.success())
+        {
+            return OsString::from(String::from(tool));
+        }
+    }
+    critical!("No container tool could be detected.");
+    std::process::exit(1);
+}
+
+fn dockercmd<'a>(args: impl IntoIterator<Item = &'a str>) -> Command {
```

Comment:
Minor nit that it's not a _docker_cmd, but 🤷 don't have a better name.

Reply:
Yeah, I wasn't totally happy about that, but 🤷🏻 too.